repositories/RoutingRepository.js

```diff
@@ -0,0 +1,169 @@
+const { head, invert, map } = require("lodash/fp");
+const mapping = {
+  QuestionPageId: "questionPageId",
+  ElseDestination: "elseDestination",
+  RoutingRuleSetId: "routingRuleSetId",
+  RuleDestination: "ruleDestination",
+  RoutingRuleId: "routingRuleId",
+  AnswerId: "answerId",
+  OptionId: "optionId",
+  ConditionId: "conditionId"
+};
+const mapFields = require("../utils/mapFields");
+const fromDb = mapFields(mapping);
+const toDb = mapFields(invert(mapping));
+
+const db = require("../db");
+
+const {
+  updateConditions,
+  createOrRemoveValue
+} = require("./strategies/routingStrategy");
+
+const Routing = require("../db/Routing");
+
+module.exports.findRoutingRuleSetByQuestionPageId = function findRoutingRuleSetByQuestionPageId(
+  where = {}
+) {
+  return Routing.findAllRoutingRuleSets()
+    .where({ isDeleted: false })
+    .where(where)
+    .first()
+    .then(fromDb);
+};
+
+module.exports.findAllRoutingRules = function findAllRoutingRules(where = {}) {
+  return Routing.findAllRoutingRules()
+    .where({ isDeleted: false })
+    .where(where)
+    .then(map(fromDb));
+};
+
+module.exports.findAllRoutingConditions = function findAllRoutingConditions(
+  where = {}
+) {
+  return Routing.findAllRoutingConditions()
+    .where(where)
+    .then(map(fromDb));
+};
+
+module.exports.findAllRoutingConditionValues = function findAllRoutingConditionValues(
+  where = {}
+) {
+  return Routing.findAllRoutingConditionValues().where(where);
+};
+
+module.exports.insertRoutingRuleSet = function insertRoutingRuleSet({
```

Comment:
Something that just occurred to me. We probably need some kind of check here to guard against being able to supply an else destination that is not in the available routing destinations. I know we'll limit the options available to the author in the UI, but we should probably check for this at the API level as well.

e.g.
```js
if (elseDestination != getAvailableRoutingDestinations()) {
	throw new Error(`Unable to route from ${questionPageId} to ${elseDestination}`);
}

Reply:
Also probably need some similar guard for the `goto` field of RoutingRule.